### PR TITLE
JS: Add syntax examples to DataFlow classes

### PR DESCRIPTION
**Pull request by asger-semmle**
Adds syntax examples to some data flow classes and predicates where it makes sense. I omitted these for `Node` and `SsaDefinitionNode` as those concepts aren't easily explained using syntax examples.

I also took the liberty of rephrasing the documentation for `ModuleImportNode` as I felt it was drawing a little too much attention to some technicalities regarding `default` imports.

### CODE REVIEW

File: javascript/ql/src/semmle/javascript/dataflow/DataFlow.qll [hunk 1]
```diff
@@ -512,6 +522,22 @@ module DataFlow {
 
   /**
    * A data flow node that writes to an object property.
+   *
+   * For example, all of the following are property writes:
+   * ```js
+   * obj.f = value;  // assignment to a member
```
Comment (max-schaefer) [inline javascript/ql/src/semmle/javascript/dataflow/DataFlow.qll]: Why "member" and not "property"?

File: javascript/ql/src/semmle/javascript/dataflow/DataFlow.qll [hunk 2]
```diff
@@ -512,6 +522,22 @@ module DataFlow {
...
+   * obj[e] = value; // assignment to a computed member
+   * { f: value }    // property initializer
+   * { g() {} }      // object literal method
+   * { get g() {}, set g(v) {} }  // accessor methods (has no rhs value)
```
Comment (max-schaefer) [inline javascript/ql/src/semmle/javascript/dataflow/DataFlow.qll]:
```suggestion
   * { get g() {}, set g(v) {} }  // accessor methods (have no rhs value)
```

File: javascript/ql/src/semmle/javascript/dataflow/Nodes.qll [hunk 3]
```diff
@@ -8,12 +8,28 @@ private import javascript
 private import semmle.javascript.dependencies.Dependencies
 private import internal.CallGraphs
 
-/** A data flow node corresponding to an expression. */
+/**
+ * A data flow node corresponding to an expression. 
+ *
+ * Example:
```
Comment (max-schaefer) [inline javascript/ql/src/semmle/javascript/dataflow/Nodes.qll]:
```suggestion
 * Examples:
```

File: javascript/ql/src/semmle/javascript/dataflow/DataFlow.qll (hunk 1 above)
Comment (asger-semmle) [reply javascript/ql/src/semmle/javascript/dataflow/DataFlow.qll]: It just felt a bit cyclic, but maybe property is better to consistency

Comment (asger-semmle) [reply javascript/ql/src/semmle/javascript/dataflow/DataFlow.qll]: Replaced with `property`

File: javascript/ql/src/semmle/javascript/dataflow/DataFlow.qll [hunk 4]
```diff
@@ -724,6 +750,17 @@ module DataFlow {
 
   /**
    * A data flow node that reads an object property.
+   *
+   * For example, all the following are property reads:
+   * ```js
+   * obj.f               // member access
+   * obj[e]              // computed member access
```
Comment (max-schaefer) [inline javascript/ql/src/semmle/javascript/dataflow/DataFlow.qll]: These two still say "member". Perhaps "property access expression"?

Comment (asger-semmle) [reply javascript/ql/src/semmle/javascript/dataflow/DataFlow.qll]: I think I prefer `property access` and `computed property access`.

**Pull request merged**